dmoj/cptbox/ptdebug.cpp

```diff
@@ -84,9 +102,67 @@ void pt_debugger::poke_reg(int idx, long data) {
 typedef int ptrace_read_t;
 #else
 typedef long ptrace_read_t;
+
+#ifndef SYS_process_vm_readv
+#define SYS_process_vm_readv 270
+#endif
+
+ssize_t __attribute__((weak)) process_vm_readv(
+    pid_t pid, const struct iovec *lvec, unsigned long liovcnt,
+    const struct iovec *rvec, unsigned long riovcnt, unsigned long flags
+) {
+    return syscall(SYS_process_vm_readv, (long) pid, lvec, liovcnt, rvec, riovcnt, flags);
+}
 #endif
 
 char *pt_debugger::readstr(unsigned long addr, size_t max_size) {
+#if PTBOX_FREEBSD
+    return readstr_peekdata(addr, max_size);
+#else
+    static unsigned long page_size = -sysconf(_SC_PAGESIZE);
+    static unsigned long page_mask = (unsigned long) page_size;
```

Comment:
Doesn't this mean `page_mask == page_size`, or is something sketchy going on here

Reply:
Oh shit this is a bug.